### DISCUSSION TITLE
`adsninja.ca` partially blocked

### DISCUSSION THREAD

**Discussion by Rudxain** [uBlockOrigin/uBlock-issues#3366] [Q&A]
I visited `https://www.xda-developers.com/*` (where `*` is the blog-post name) on Firefox 129 for Android, `cdn.adsninja.ca` wasn't completely blocked.

- **all** filter lists enabled (Both AdGuard and EasyList have lines explicitly containing `adsninja.ca`)
- Medium Mode
- `no-scripting: * true` (logger not needed)

I've found this source-code from the blog page (I removed indentation and extra `\n`):
```html
<link rel='preload' href='https://cdn.adsninja.ca/adsninja_client_style.css?v=934' as='style' onload="this.onload=null;this.rel='stylesheet'">
<noscript><link rel='stylesheet' href='https://cdn.adsninja.ca/adsninja_client_style.css?v=934'></noscript>
<!-- Ad Header End -->
<!-- GLOABL JS VARIABLES -->
```

> Yes, the comment says "GLOABL" (typo). But it seems to refer to code below it.

I suspect 1 or more stylesheets went through, that's why I included the snippet.

I know it's possible for a static filter to override another, thereby allowing some requests. But why did it happen here?

**Comment by stephenhawk8054** [comment]
Medium mode and `no-scripting` don't block stylesheet connections.

This is the stylesheet connection that will load with your configuration:

```html
<noscript><link rel='stylesheet' href='https://cdn.adsninja.ca/adsninja_client_style.css?v=934'></noscript>
```

**Reply by Rudxain** [reply]
> Medium mode and `no-scripting` don't block stylesheet connections.

That's obvious to me. I just said that to clarify that the logger is not necessary (but it is convenient) to debug this

**Reply by stephenhawk8054** [reply]
I don't really understand your question.

> But why did it happen here?

What is "it"?

---

Just FYI that EL has a whitelist filter for `adsninja.ca` in this site:

https://github.com/easylist/easylist/blob/30595d2e14e41535a7d754aa80a11e8420fdc9cc/easylist/easylist_allowlist.txt#L80

**Reply by Rudxain** [reply]
> What is "it"?

When I wrote that, I assumed that some unknown filter blocks all subdomains of `adsninja.ca`, while another filter allowed the specific sub `cdn.` (for all or some domains that request it).

By "it" I meant "some static filter allows some requests to `cdn.adsninja.ca`".

However, I was open to the possibility that there's another problem.

> Just FYI that EL has a whitelist filter for `adsninja.ca` in this site:
> 
> https://github.com/easylist/easylist/blob/30595d2e14e41535a7d754aa80a11e8420fdc9cc/easylist/easylist_allowlist.txt#L80

Thanks for the info! I've read that line before, but didn't know the semantics of `@@` prefix
